src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -79,14 +80,40 @@ public ServiceManagerBuilder WithNewtonsoftJson(Action<NewtonsoftServiceHubProto
         }
 
         /// <summary>
-        /// Uses Newtonsoft.Json library to serialize messages sent to SignalR.
+        /// Uses Newtonsoft.Json library to serialize messages sent to SignalR clients.
         /// </summary>
         /// <returns>The <see cref="ServiceHubContextBuilder"/> instance itself.</returns>
         public ServiceManagerBuilder WithNewtonsoftJson()
         {
             return WithNewtonsoftJson(o => { });
         }
 
+        /// <summary>
+        /// Sets the SignalR hub protocols to serialize messages sent to SignalR clients.
+        /// </summary>
+        /// <remarks><para>Currently it only works for <b>persistent</b> mode. The support for <b>transiet(default)</b> mode is to be done. </para> 
+        /// <para>Calling this method clears the default hub protocol.</para></remarks>
+        /// <param name="hubProtocols">Only the protocols named "json" or "messagepack" are allowed.</param>
+        /// <returns>The <see cref="ServiceHubContextBuilder"/> instance itself.</returns>
+        public ServiceManagerBuilder WithHubProtocol(IEnumerable<IHubProtocol> hubProtocols)
+        {
+            if (hubProtocols == null)
+            {
+                throw new ArgumentNullException(nameof(hubProtocols));
+            }
+            // Allows the user to use MessagePack only.
+            _services.RemoveAll<IHubProtocol>();
```

Comment:
shall we move this check to ServiceManagerBuilder.Build()

Reply:
Why do we want to move the check to `ServiceManagerBuilder.Build()`?

Reply:
Build() is the time when everything is set. What if IHubProtocol is added after `WithHubProtocol`?